### ISSUE TITLE
reviews to resolve

### ISSUE THREAD

**Issue by smilesun** [compstat-lmu/paper_2019_variationalResampleDistributionShift#40]
SUBMISSION: 36
TITLE: Resampling-based Assessment of Robustness to Distribution Shift for Deep Neural Networks

----------------------- REVIEW 1 ---------------------
SUBMISSION: 36
TITLE: Resampling-based Assessment of Robustness to Distribution Shift for Deep Neural Networks
AUTHORS: Xudong Sun, Yu Wang, Alexej Gossman and Bernd Bischl

----------- Positive Aspects of the Paper -----------
The authors propose an interesting idea on how to assess the robustness CNN (possibly extendible to other architectures). The approach is quite elegant and does not require other datasets than those originally given. 

The paper is well written and good to understand.
----------- Negative Aspects of the paper -----------
There are contradictory results in the experiments. The training accuracies for the original data split (Table 2) are significantly better than for the ordinary “CV” experiments. The training datasets should have the same distributions here, but are 20% smaller in size for “CV”. This should potentially lead to more overfitting, hence higher training accuracy in the "CV" experiments. The experiments and Table 3 show the opposite behaviour. The cause of this behaviour must be examined in the paper, otherwise it could also be interpreted as an indicator for a bug in the experimental setup.

Minor issues:
- please provide details about the used architecture of the VAE
- please provide details about the used architecture of the VGMM
   - how is it provided that all splits have the same cardinality, which is important to maintain the same $p(y)$ in $D_k$?
   - a visualisation of samples from different $d_k^c$ would be very interesting
- Figure 1 appears rather crowded, do k=1...5 panels or (better) c=1…10 panels showing the tSNE embeddings for $d_k^c$ individually next to each other
----------- Overall evaluation -----------
SCORE: 0 (borderline paper)
----- TEXT:
I see the contradictory training accuracies mentioned above critical for the acceptance of the paper.
If this issue (plus the minor ones) were resolved (or at least addressed) I could mend my overall evaluation to "2: accept".

----------------------- REVIEW 2 ---------------------
SUBMISSION: 36
TITLE: Resampling-based Assessment of Robustness to Distribution Shift for Deep Neural Networks
AUTHORS: Xudong Sun, Yu Wang, Alexej Gossman and Bernd Bischl

----------- Positive Aspects of the Paper -----------
1. interesting problem addressed: covariate shift
  2. interesting method proposed: plant it, but still use observed instances only
  3. well written paper
----------- Negative Aspects of the paper -----------
1. the specific planting method is not well motivated.
  2. the relation to transfer learning and models for detecting covariate shift is not made explicit.
  3. the quality criteria for a good covariate shift planting method are not clear.
----------- Overall evaluation -----------
SCORE: 0 (borderline paper)
----- TEXT:
The paper addresses the problem of transfer learning, esp.
when train and test samples vary by covariate shift. The authors
propose a novel train/test split procedure to deliberately
create covariate shift when there is none in the original data.
To do so, they cluster the instances by their latent features of
an autoencoder and split train/test by cluster, always assigning
one cluster to test and all others to train, stratified by target.
In experiments with deep neural networks and an image dataset
they show that this procedure actually creates covariate shift
and that the performance of neural networks deteriorates when
exposed to it.

The problem addressed is interesting, the paper well structured
and well written. Esp. interesting is that the method of the authors
plants covariate shift by a mere non-uniform sampling procedure
and not by transforming the covariates which easily can introduce
unrealistic artifacts, e.g., when adding pixelwise noise to images.

The authors do not provide much motivation why they think the specific
way they plant covariate shift is relevant: do similar processes
take place in reality, e.g., that covariates in data of similar tasks
decompose into clusters? Without such a motivation, any other, possibly
way simpler procedure such as the typical k-means initialization
strategy might do the job as well.

Evaluating models on tasks/distributions they have not been trained on
to me looks like an ill-posed problem: there is absolutely no reason the
models should do anything useful. The typical setting in machine learning
is slightly different: in transfer learning the model will get
access to some, few instances of the target distribution and the goal
is to learn from these instances a better model transfering from a
similar task than from the data of the target task alone. This work
is not discussed, see, e.g. [1].

[1] Weiss, K., Khoshgoftaar, T.M., Wang, D., 2016. A survey of
  transfer learning. Journal of Big Data 3, 9.

----------------------- REVIEW 3 ---------------------
SUBMISSION: 36
TITLE: Resampling-based Assessment of Robustness to Distribution Shift for Deep Neural Networks
AUTHORS: Xudong Sun, Yu Wang, Alexej Gossman and Bernd Bischl

----------- Positive Aspects of the Paper -----------
+testing and even breaking ML models is important
+ novel resampling technique based on deep models
----------- Negative Aspects of the paper -----------
- all models show strong effects due to a shift
- not clear whether we can also control the magnitude of the shift easily.
----------- Overall evaluation -----------
SCORE: 1 (weak accept)
----- TEXT:
The paper extends restrictive federated model selection over shifted
distribution to computer vision tasks. To this end, it proposes a resampling framework 
to assess the robustness to distribution shift of deep learning models.

Overall the paper is well written and structure. Here and there it is probably a little bit too much to the point as e.g. in Section 2, which is very short for a general AI audience. Nevertheless, the idea of using an idea similar to cross validation but now for generating bins with shifting distributions is nice and indeed helps to  characterize the deep learning model's ability to generalize under distribution shift. Even more importantly, the empirical evaluation is well design and touches interesting question. However, there is also a major downside that should be addressed. Somehow you state that you are surprised that Bayesian CNNs are also affected by the distribution shift. Why? The distributions shifts you are generating are rather strong, going from one cluster to another. Put it differently, the statements that "even though BCNNs are known to posses superior generalization properties" may to fit here well as your shifts are simply too strong. Is there away in you !
 resampling approach to control the magnitude of the shift? This is also important in order to show that you method is indeed valuable. Is there a model that could deal with the shift produced?
